File: client/src/interfaces/Ingredient.ts
export interface IngredientListItem {
  ingredient_id: string;
  name?: string;
  image_path?: string;
  inbound_at?: string;
  expiration_at?: string;
}

export interface IngredientListResponse {
  content: IngredientListItem[];
  sliceInfo: {
    currentPage: number;
    pageSize: number;
    hasNext: boolean;
  };
}

export interface DetailedIngredient extends IngredientListItem {
  member_id: string;
  member_name: string;
  member_image_path: string;
}

export type Ingredient = IngredientListItem | DetailedIngredient;

export default Ingredient;

File: mobile/SeethroughApp/app/src/main/java/com/example/seethroughapp/model/ingredient/Ingredient.java
package com.example.seethroughapp.model.ingredient;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter @Setter
@ToString
public class Ingredient {
    private String ingredientId;
    private String name;
    private String imagePath;
    private String inboundAt;
    private String expirationAt;
}
